test/functional/feature_blockfilterindex_prune.py

```diff
@@ -18,32 +18,41 @@ def set_test_params(self):
     def run_test(self):
         # test basic pruning compatibility & filter access of pruned blocks
         self.log.info("check if we can access a blockfilter when pruning is enabled but no blocks are actually pruned")
-        assert len(self.nodes[1].getblockfilter(self.nodes[1].getbestblockhash())['filter']) > 0
+        self.sync_all()
```

Comment:
sync_all syncs the mempool and blocks, but not the blockfilter. You'll need `self.wait_until(lambda: self.nodes[1].getindexinfo() == {'basic block filter index': {'synced': True, 'best_block_height': 200}})`. Either in this line or by modifying sync_all.

Reply:
Thanks! I hadn't seen #21252 that fixes the issue.